docs/configuring.md

```diff
@@ -181,6 +186,45 @@ JupyterLite instances hosted there. To create separate settings stores, change t
 `jupyter-lite.json#jupyter-config-data/settingsStorageName` from the default of
 `JupyterLite Storage`.
 
+By default, the best available, persistent storage driver will be used. One may force a
+particular set of drivers to try with
+`jupyter-lite.json#jupyter-config-data/settingsStorageDrivers`. See more about
+[local storage drivers](#local-storage-drivers).
+
+## Local Storage Drivers
+
+By default, the "best" [localForage] driver will be selected from the technologies
+available in the user's browser.
+
+[localforage]: https://github.com/localForage/localForage
+
+To force choosing from a particular set of technologies, `settingsStorageDrivers` and
+`contentsStorageDrivers` can be specified, with the first browser-compatible driver
+being chosen.
+
+| configuration value   | technology   | persistent? | note                                   |
+| --------------------- | ------------ | ----------- | -------------------------------------- |
+| `asyncStorage`        | IndexedDB    | yes         | usually the one selected               |
+| `webSQLStorage`       | WebSQL       | yes         |                                        |
+| `localStorageWrapper` | localStorage | yes         |                                        |
+| `memoryStorageDriver` | in-memory    | **NO**      | requires `enableMemoryStorage`         |
+| _other_               | _unknown_    | _unknown_   | may be added by third-party extensions |
+
+### Volatile Memory Storage
+
+Many extensions and features require the ability to at least _think_ they are saving and
+loading contents and settings. If a user's data cannot be stored due to browser security
+settings, a JupyterLite app will generally fail to fully initialize: while this might be
+frustrating, losing a user's unique data creation is _even more_ frustating.
+
+```{warning}
+If persistence is **entirely** handled outside of JupyterLite, e.g. in an embedded
+[`repl`](./applications/repl.md) it is possible to disable all persistence, assuring
```

Comment:
The `repl` app was already not persisting data to the browser since it's based on the ephemeral lab code console.

But agree switching to an in-memory storage for this app will help to eventually allow for other widgets (for example https://github.com/jupyterlite/jupyterlite/issues/148) :+1:

Reply:
> not persisting data to the browser

it _would_ persist settings

Reply:
> it would persist settings

Ah right sure, for example themes and other user settings.

Reply:
right: the joke on #525 was, i tried to open an iframe to `repl` from inside a notebook (hosted on the same site), and by default a lot of stuff was broken... the memory fallback enables a way to require slightly fewer sandbox exceptions to work "properly,"